tests/e2e/utils/auto-rds-s3-setup.py

```diff
@@ -176,141 +180,111 @@ def does_secret_already_exist(secrets_manager_client, secret_name):
-        "rds",
-        region_name=CLUSTER_REGION
-    )
+        print(
+            f"Skipping RDS setup, DB instance '{DB_INSTANCE_NAME}' already exists!")
 
 
 def does_database_exist(rds_client):
     matching_databases = rds_client.describe_db_instances(
-        Filters=[
-            {
-                "Name": "db-instance-id",
-                'Values': [DB_INSTANCE_NAME]
-            }
-        ]
+        Filters=[{"Name": "db-instance-id", "Values": [DB_INSTANCE_NAME]}]
     )["DBInstances"]
 
     return len(matching_databases) > 0
 
 
-def setup_db_instance(rds_client):
-    setup_db_subnet_group(rds_client)
-    return create_db_instance(rds_client)
+def setup_db_instance(rds_client, secrets_manager_client, eks_client, ec2_client):
+    setup_db_subnet_group(rds_client, eks_client, ec2_client)
+    return create_db_instance(
+        rds_client, secrets_manager_client, eks_client, ec2_client
+    )
 
 
-def setup_db_subnet_group(rds_client):
+def setup_db_subnet_group(rds_client, eks_client, ec2_client):
     if not does_db_subnet_group_exist(rds_client):
-        create_db_subnet_group(rds_client)
+        create_db_subnet_group(rds_client, eks_client, ec2_client)
     else:
-        print(f"Skipping DB subnet group creation, DB subnet group '{DB_SUBNET_GROUP_NAME}' already exists!")
+        print(
+            f"Skipping DB subnet group creation, DB subnet group '{DB_SUBNET_GROUP_NAME}' already exists!"
+        )
 
 
 def does_db_subnet_group_exist(rds_client):
     try:
-        rds_client.describe_db_subnet_groups(DBSubnetGroupName=DB_SUBNET_GROUP_NAME)
+        rds_client.describe_db_subnet_groups(
+            DBSubnetGroupName=DB_SUBNET_GROUP_NAME)
         return True
     except rds_client.exceptions.DBSubnetGroupNotFoundFault:
         return False
 
 
-def create_db_subnet_group(rds_client):
+def create_db_subnet_group(rds_client, eks_client, ec2_client):
     print("Creating DB subnet group...")
 
-    subnet_ids = get_cluster_private_subnet_ids()
+    subnet_ids = get_cluster_private_subnet_ids(eks_client, ec2_client)
 
     rds_client.create_db_subnet_group(
         DBSubnetGroupName=DB_SUBNET_GROUP_NAME,
         DBSubnetGroupDescription="Subnet group for Kubeflow metadata db",
-        SubnetIds=subnet_ids
+        SubnetIds=subnet_ids,
     )
 
     print("DB subnet group created!")
 
 
-def get_cluster_private_subnet_ids():
-    ec2_client = get_ec2_client()
+def get_cluster_private_subnet_ids(eks_client, ec2_client):
+    subnet_ids = eks_client.describe_cluster(name=CLUSTER_NAME)["cluster"][
+        "resourcesVpcConfig"
+    ]["subnetIds"]
 
-    subnets = ec2_client.describe_subnets(
-        Filters=[
-            {
-                "Name": "tag:alpha.eksctl.io/cluster-name",
-                "Values": [CLUSTER_NAME]
-            },
-            {
-                "Name": "tag:aws:cloudformation:logical-id",
-                "Values": ["SubnetPrivate*"]
-            },
-        ]
-    )["Subnets"]
```

Comment:
using the filter avoids having to paginate through the describe_subnets results


filtering in memory could result in an empty set depending on the first batch of returned results

Reply:
This filter relies on `     "Name": "tag:alpha.eksctl.io/cluster-name",
                "Values": [CLUSTER_NAME]` which as mentioned can cause issues where clusters share vpc/subnets. Here we are getting the subnets from the describe cluster command and then only describing those subnets from the describe_subnets() command.